kernel/names.ml

```diff
@@ -495,14 +495,19 @@ module KerPair = struct
     | Dual (kn,_) -> kn
 
   let same kn = Same kn
-  let make knu knc = if KerName.equal knu knc then Same knc else Dual (knu,knc)
+  let make_pair knu knc = if KerName.equal knu knc then Same knc else Dual (knu,knc)
+
+  let of_kn ?user knc = match user with None -> same knc | Some knu -> make_pair knu knc
+  let make ?user mpc l =
+    let knc = KerName.make mpc l in
+    match user with None -> same knc | Some mpu -> make_pair (KerName.make mpu l) knc
 
-  let make1 = same
-  let make2 mp l = same (KerName.make mp l)
-  let repr2 kp = KerName.repr (user kp)
   let label kp = KerName.label (user kp)
   let modpath kp = KerName.modpath (user kp)
 
+  let canonize kp = Same (canonical kp)
```

Comment:
These functions should be part of the `Environ.QConstant` and related API instead. The goal is to impose to have an environment around to make sense of a dual name, and eventually replacing them with a single name + a fetch in the environment.